### PR TITLE
Finish testing crank

### PR DESCRIPTION
**Pull request by andrewgordstewart**
This finishes testing `crank` about as well as the old unit tests. There are three different test scenarios: as Alice, as Bob and as P1, in three different functions. This creates *some* duplicated code, but note that the expected behaviour depends on:
- whose perspective we're testing (Alice, Bob or P1)
- what roles they play in their ledger channel (Leader, Follower)

This makes it a little awkward to write a single `TestCrankAsActor` function that tests a good subset of the possible values. Instead,
- `TestCrankAsP1` has one leader channel and one follower channel, which checks that 
- `TestCrankAsAlice` has a leader channel
- `TestCrankAsBob` has a follower channel

To make the tests pass, `IsProposed` can be called by a follower -- the virtualfund objective uses it to detect whether or not it should accept a ledger update when its ledger channel is a follower channel.

#### Code quality

- [ ] This change does not have an unduly wide scope
... yes, apologies!

### CODE REVIEW

File: protocols/virtualfund/virtualfund_single_hop_test.go [hunk 1]
```diff
@@ -205,30 +220,186 @@ func TestCrankAsAlice(t *testing.T) {
...
+	postFS := state.NewSignedState(o.V.PostFundState())
+	mySig, _ = postFS.State().Sign(my.privateKey)
+	_ = postFS.AddSignature(mySig)
+
+	ok(t, err)
+	equals(t, waitingFor, WaitingForCompletePostFund)
+	assertStateSentTo(t, effects, postFS, bob)
+}
+
+// TestCrankAsBob tests the behaviour from a end-user's point of view when they are a follower in the ledger channel
```
Comment (NiloCK) [inline protocols/virtualfund/virtualfund_single_hop_test.go:264]: This comment (along with the similar one for `TestCrankAsAlice`) seems to tie the vfo role `bob` to the `consensuschannel` role `follower`, but these are unrelated ideas.

**Review by NiloCK — APPROVED** [review]
This LGTM.

File: protocols/virtualfund/virtualfund.go [hunk 2]
```diff
@@ -592,14 +592,12 @@ func (o *Objective) proposeLedgerUpdate(connection Connection, sk *[]byte) (prot
 // acceptLedgerUpdate checks for a ledger state proposal and accepts that proposal if it satisfies the expected guarantee.
 func (o *Objective) acceptLedgerUpdate(c Connection, sk *[]byte) (protocols.SideEffects, error) {
 	ledger := c.Channel
-	err := ledger.SignNextProposal(c.expectedProposal(), *sk) // todo: #420 -- need to send side effects!
+	signedProposal, err := ledger.SignNextProposal(c.expectedProposal(), *sk) // todo: #420 -- need to send side effects!
```
[the remark this answers is not in the document]
Comment (andrewgordstewart) [reply protocols/virtualfund/virtualfund.go:595]: good catch!

File: protocols/virtualfund/virtualfund_single_hop_test.go (hunk 1 above)
Comment (andrewgordstewart) [reply protocols/virtualfund/virtualfund_single_hop_test.go:264]: Yes, good point. I am fixing these in a follow-on: https://github.com/statechannels/go-nitro/pull/530 (just doing a self-review now)

File: protocols/virtualfund/virtualfund.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrewgordstewart) [reply protocols/virtualfund/virtualfund.go:595]:
```suggestion
	signedProposal, err := ledger.SignNextProposal(c.expectedProposal(), *sk)
```
